test/ResourceNoManualCurrencyFormatting.test.js

```diff
@@ -0,0 +1,275 @@
+/*
+ * ResourceNoManualCurrencyFormatting.test.js - test the built-in regular-expression-based rules
+ *
+ * Copyright © 2023 JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ResourceString, ResourceArray, ResourcePlural } from 'ilib-tools-common';
+
+import ResourceSourceChecker from '../src/rules/ResourceSourceChecker.js';
+import { regexRules } from '../src/plugins/BuiltinPlugin.js';
+
+import { Result, IntermediateRepresentation } from 'i18nlint-common';
+
+function findRuleDefinition(name) {
+    return regexRules.find(rule => rule.name === name);
+}
+
+describe("testResourceNoManualCurrencyFormatting", () => {
+    test("ResourceManualCurrencyFormatting", () => {
+        expect.assertions(8);
+
+        const rule = new ResourceSourceChecker(findRuleDefinition("source-no-manual-currency-formatting"));
+        expect(rule).toBeTruthy();
+
+        const resource = new ResourceString({
+            key: "matcher.test",
+            sourceLocale: "en-US",
+            source: "It costs ${num}.",
```

Comment:
Do any frameworks we use by default rely on this syntax for replacement parameters? If so, we could run into some false-positives

Reply:
Oh crap, I think maybe python strings have that syntax...

Reply:
As discussed during the call, the easy solution is to either move this into React plugin (and then replicate it everywhere else except Python), or have Python plugin automatically disable this rule.

The hard solution would be for all Resources to allow an AST and output that from resource file parsers (i.e. add an abstraction similar to how we do with React plurals).

Reply:
Okay, I checked it out. Python uses:

- {this} syntax
- %(this)s syntax
- and $this syntax

for string interpolation. The only languages that use ${this} syntax for string interpolation are Javascript template strings, and groovy template strings.

So this PR does not have a conflict right now. (We don't really need to support groovy for now.)